src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Helpers.cs

```diff
@@ -12,54 +12,113 @@ namespace System.Text.Json
 {
     public sealed partial class Utf8JsonWriter
     {
-        private bool HasPartialStringData => PartialStringDataLength != 0;
+        /// <summary>
+        /// Assuming that the writer is currently in a valid state, this returns true if a JSON value is allowed at the current position.
+        /// <list type="bullet">
+        /// <item>
+        /// If <see cref="_enclosingContainer"/> is an array then writing a value is always allowed.
```

Comment:
Seems easier to read if you just quote the enum value directly
```suggestion
        /// If it's <see cref="EnclosingContainerType.Array"/> then writing a value is always allowed.
```